src/Sentry.Unity/Integrations/UnityLogHandlerIntegration.cs

```diff
@@ -1,54 +1,102 @@
 using System;
 using Sentry.Extensibility;
 using Sentry.Integrations;
+using Sentry.Protocol;
 using UnityEngine;
 
 namespace Sentry.Unity.Integrations
 {
-    internal sealed class UnityApplicationLoggingIntegration : ISdkIntegration
+    internal sealed class UnityLogHandlerIntegration : ISdkIntegration, ILogHandler
     {
         internal readonly ErrorTimeDebounce ErrorTimeDebounce = new(TimeSpan.FromSeconds(1));
         internal readonly LogTimeDebounce LogTimeDebounce = new(TimeSpan.FromSeconds(1));
         internal readonly WarningTimeDebounce WarningTimeDebounce = new(TimeSpan.FromSeconds(1));
 
-        // TODO: remove 'IEventCapture' in  further iteration
-        private readonly IEventCapture? _eventCapture;
         private readonly IApplication _application;
 
         private IHub? _hub;
         private SentryUnityOptions? _sentryOptions;
 
-        public UnityApplicationLoggingIntegration(IApplication? application = null, IEventCapture? eventCapture = null)
+        private ILogHandler _unityLogHandler = null!; // Set during register
+
+        public UnityLogHandlerIntegration(IApplication? application = null)
         {
             _application = application ?? ApplicationAdapter.Instance;
-            _eventCapture = eventCapture;
         }
 
         public void Register(IHub hub, SentryOptions sentryOptions)
         {
             _hub = hub;
             _sentryOptions = sentryOptions as SentryUnityOptions;
 
-            _application.LogMessageReceived += OnLogMessageReceived;
+            _unityLogHandler = Debug.unityLogger.logHandler;
+            Debug.unityLogger.logHandler = this;
+
             _application.Quitting += OnQuitting;
         }
 
-        // Internal for testability
-        internal void OnLogMessageReceived(string? condition, string? stackTrace, LogType type)
+        public void LogException(Exception exception, UnityEngine.Object context)
+        {
+            CaptureException(exception, context);
+            _unityLogHandler.LogException(exception, context);
+        }
+
+        internal void CaptureException(Exception exception, UnityEngine.Object? context)
```

Comment:
what's in `context`? Anything useful to add to the event or crumb?

Reply:
That's the crazy awesome part: When you're calling `Debug.Log("My log message", this);` you can actually pass "whatever" in there as context (the MonoBehaviour in this case). Inside the editor this lets you click the log in the console and highlights the GameObject in the hierarchy. What that means for us is that we could grab all kinds of useful stuff from there.

- The name of the GameObject 
- Other components on that GameObject
- Child Count
- e.t.c.

I'm just not sure what to grab. Probably the name.